src/synthesiser/SynthesiserRelation.cpp

```diff
@@ -213,9 +223,20 @@ void SynthesiserDirectRelation::generateTypeStruct(std::ostream& out) {
             out << "  return ";
             std::function<void(size_t)> genless = [&](size_t i) {
                 size_t attrib = ind[i];
-                out << " a[" << attrib << "] < b[" << attrib << "]";
+
+                std::string element_type_cast("");
+
+                switch (types[attrib][0]) {
+                    case 'i': element_type_cast = "(RamSigned)"; break;
+                    case 'u': element_type_cast = "(RamUnsigned)"; break;
+                    case 'f': element_type_cast = "(RamFloat)"; break;
```

Comment:
Are these correct or should the be "ramBitCast<type>("?

Reply:
Fixed!